### PR TITLE
Thugs 1730 sparkplug sends heartbeats

### PR DESCRIPTION
**Pull request by drone115b**
Send heartbeats at regular intervals.

This implementation will fail if the time to process a message exceeds the heartbeat expiry duration.

### CODE REVIEW

File: sparkplug/config/connection.py [hunk 1]
```diff
@@ -59,28 +59,33 @@ def __init__(self, name, channel_configurer, reconnect_delay='10', **kwargs):
...
             raise
 
-    def pump(self, channel):
+    def pump(self, connection, channel):
         while True:
             _log.debug("Waiting for a message.")
-            channel.wait(spec.Basic.Deliver)
+            try:
+                channel.wait(spec.Basic.Deliver, timeout=connection.heartbeat)
+            except socket.timeout :
```
Comment (tobioboye) [inline sparkplug/config/connection.py:77]: From the looks, this will swallow all socket timeouts, right? It it worth logging the exception?

**Comment by tobioboye** [comment]
Overall solid changes! 👏 I think this will reduce the number of TCP errors we get!

File: sparkplug/config/connection.py (hunk 1 above)
Comment (drone115b) [reply sparkplug/config/connection.py:77]: Sorry, I missed this comment before I merged.   I'm not sure what our temperature is for logging volume (i.e. will show a timeout every connection.heartbeat seconds), but I can go back and add that logger in if you think it's important.

Comment (tobioboye) [reply sparkplug/config/connection.py:77]: Actually, there's no need to log it since the exception is as a result of the timeout waiting on the channel.

**Pull request merged**